File: paddler_client_python/paddler_client/types/tool.py
from typing import Literal

from pydantic import BaseModel

from paddler_client.types.validated_parameters_schema import (
    ValidatedParametersSchema,
)


class Function(BaseModel):
    name: str
    description: str
    parameters: ValidatedParametersSchema | None = None


class Tool(BaseModel):
    type: Literal["function"] = "function"
    function: Function

File: paddler_client_javascript/src/schemas/Tool.ts
import { z } from "zod";

import { ValidatedParametersSchemaSchema } from "./ValidatedParametersSchema";

export const FunctionDefinitionSchema = z.object({
  name: z.string(),
  description: z.string(),
  parameters: ValidatedParametersSchemaSchema.optional(),
});

export const FunctionCallToolSchema = z.object({
  type: z.literal("function"),
  function: FunctionDefinitionSchema,
});

export const ToolSchema = FunctionCallToolSchema;

export type FunctionDefinition = z.infer<typeof FunctionDefinitionSchema>;
export type FunctionCallTool = z.infer<typeof FunctionCallToolSchema>;
export type Tool = z.infer<typeof ToolSchema>;
